modules/gsuite_enabled/main.tf

```diff
@@ -67,7 +65,7 @@ resource "gsuite_group_member" "api_s_account_api_sa_group_member" {
 module "project-factory" {
   source = "../core_project_factory/"
 
-  group_email                 = "${module.gsuite_group.email}"
+  group_email                 = "${gsuite_group.group.email}"
```

Comment:
I don't think this logic will work for cases where `create_group = false`. I think we might be better off with an explicit `depends_on` to solve this issue.

Reply:
Good catch. I just tested that scenario and you are correct about there being a problem when `create_group = false`.

Unfortunately, Terraform doesn't support depending on a module yet so that's not an option to fix the race. The `gsuite_group` resource needs to be referenced at some point in the graph to create the implicit dependency.

That being said, I came up with a bit of a hack to get the group definition into the graph or to use the provided email if `create_group` is set to false. It appears to be working as expected.

PTAL.